src/main/kotlin/com/jetbrains/snakecharm/codeInsight/completion/SmkRulesAndCheckpointsCompletionContributor.kt

```diff
@@ -132,34 +151,24 @@ private class SmkRulesAndCheckpointsObjectsCompletionProvider : CompletionProvid
 
 // the following functions can be used to implement completion for any section/object referring to rule names
 
-private fun collectVariantsForElement(element: PsiElement): MutableList<Pair<String, SmkRuleOrCheckpoint>> {
+private fun <Psi: SmkRuleOrCheckpoint> collectVariantsForElement(
+        element: PsiElement,
+        indexKey: StubIndexKey<String, Psi>,
+        sectionClass: Class<Psi>,
+        currentFileDeclarationsFunction: (SmkFile) -> List<Pair<String, Psi>>
+): MutableList<Pair<String, SmkRuleOrCheckpoint>> {
```

Comment:
Not sure, that usage of `MutableList` here provides noticeable benefits. Immutable collections are more  readable / easy to understand immutable collections. This method force use to think who and why modifies return result of this method, track usages etc.

Reply:
done